### ISSUE TITLE
DT crashes with no reason

### ISSUE THREAD

**Issue by Alain2Bretagne** [darktable-org/darktable#17016]
### Describe the bug

DT crashes with no reason, mainly during export but not only. When I restart DT and try again, it works quite a long time and crashes with no reason

### Steps to reproduce

I don't have...

### Expected behavior

no crash...

### Logfile | Screenshot | Screencast

simply disappears. Nothing to download

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.6.1

### What OS are you using?

Linux

### What is the version of your OS?

Ubuntu 22.04.4 LTS

### Describe your system?

Mem: 16Go cpu:intel core I7-8700 no gpu

### Are you using OpenCL GPU in darktable?

None

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

It seems that the problem appears with my nex camera FX full frame, generating files up to 50 Mo per picture. I don't remerber any problem with my former camera with smaller files. Not fully sure...

**Comment by ptilopteri** [comment]
* Alain2Bretagne ***@***.***> [06-18-24 11:42]:
> ### Describe the bug
> 
> DT crashes with no reason, mainly during export but not only. When I restart DT and try again, it works quite a long time and crashes with no reason
> 
> ### Steps to reproduce
> 
> I don't have...
> 
> ### Expected behavior
> 
> no crash...

can you provide via a file sharing service a file which causes a crash and
an xmp if available?


-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by Alain2Bretagne** [comment]
while processing an image NEF, I crashed LINUX just because I was zooming in the image. I had to reboot the machine. The disk led was blinking and I suppose that it was using the disk at full speed, but in fact I don't know as nothing was reactive. The machine restarted properly and DT as well. Nothing understandable for me...

**Comment by victoryforce** [comment]
Modern operating systems should not crash no matter what. If any application contains a bug and crashes, only that application will be terminated, but other applications and the OS cannot be affected.

Therefore, if you experience freezes/reboots of the operating system, then the root cause of the problem is not in the application.

It may be a problem in one of the drivers that the program simply triggered, it may be a problem in the hardware of the computer, which appears under heavy load. Examples of what is often observed: overheating of one of the computer components (CPU, disk, GPU), a problem with the power supply unit, the power of which is not enough for peak loads.

BTW, since you are processing multi-megapixel images on just the CPU, it will be under a really heavy load.

**Comment by Alain2Bretagne** [comment]
This picture always crashes DT during exportation.

https://drive.proton.me/urls/A78XRGQYJM#0x6DbZemIX61

**Comment by victoryforce** [comment]
> This picture always crashes DT during exportation.

It's great that you provided a file on which the crash should be reproduced, this allows us to test the assumption that the crash is caused by some specifics of a particular image. No, I don't have any crashes on this image, either when exporting, when processing, or when navigating.

As I wrote above, ___the freezing of the OS itself indicates that the problem is not in darktable___. A bug in the program can cause this program to crash, but it is not able to cause the OS to freeze. If this happens, look for the problem in your system. So I'm closing this issue.

**Comment by Alain2Bretagne** [comment]
Sorry Victor, I disagree with that.

Please watch the video at the link: https://drive.proton.me/urls/A78XRGQYJM#0x6DbZemIX61

It clearly shows that DT disappears with no reason while the system is still active and safe.

I have probably said the things improperly: linux does not crash but can be so overloaded by a "crazy task" that I cannot do anything.

I have tested different processings on this picture and things are not clear but the crash happens or not according the the modules I have activated. Seems to be more a combination of modules more than one by itself.

Thank you

**Comment by ralfbrown** [comment]
Rereading this thread, it looks like misconfigured swap space.  The old recommendation of swap = 1 to 2 times RAM from the days of 4 **mega**bytes of RAM no longer holds, as it can take *minutes* to swap multiple gigabytes of data in and out.  My first-ever workstation with 16 GB RAM had been configured with 20 GB swap by IT, and one time I had a runaway program use up all RAM *and* all swap before the out-of-memory killer killed it.  The workstation was unresponsive for 20+ minutes until everything which had been swapped to spinning disk got swapped in again.

My home PCs have been setup without swap ever since I got up to 16 GB RAM, though it may be useful to have a few hundred megabytes of swap on spinning media or a gigabyte or two on SSD to allow program initialization code (which will never be executed again) to be swapped out.

**Comment by Alain2Bretagne** [comment]
Hi Ralf

You got it!

DT runs out of memory and then crashes or blocks everything in swap access.

I stopped everything (browser, mails, music player...) but DT and then it works perfectly and I could export more than one hundred pictures in one batch with no problem. In fact, I need a bit more than 16 G to work properly.

So I have to buy some more memory or extend my swap file. The problem is now fixed.

Thank you for your contribution.
